### ISSUE TITLE
I don't see the following definitions for these two functions, can you provide them，Many thanks

### ISSUE THREAD

**Issue by ZZ-626** [XiangZhang1997/EAG-SAE-ESTF#1]
hello  I don't see the following definitions for these two functions, can you provide them，Many thanks
  self.ca = ChannelAttentionafn(in_channels)
  self.sa = SpatialAttentionafn(in_channels)

**Comment by XiangZhang1997** [comment]
Thank you for your attention to our work. Here are the corresponding class modules: 

class SpatialAttentionafn(nn.Module):
    def __init__(self,channel,reduction=16):
        super(SpatialAttentionafn, self).__init__()
        self.msconv1 = nn.Sequential(
                ConvBlock(channel,channel//reduction,kernel_size=3,padding=1)
            )
            self.msconv2 = nn.Sequential(
                ConvBlock(channel,channel//reduction,kernel_size=5,padding=2)
            )
            self.msconv3 = nn.Sequential(
                ConvBlock(channel,channel//reduction,kernel_size=7,padding=3)
            )
            self.conv = nn.Sequential(
                nn.Conv2d(2, 1, kernel_size=(3, 3), padding=(1, 1)),
                nn.Sigmoid()
            )
    def forward(self, x):
        x1 = self.msconv1(x)
        x2 = self.msconv2(x)
        x3 = self.msconv3(x)
        cat = torch.cat([x1,x2,x3],dim=1)
        avg_out = torch.mean(cat, dim=1, keepdim=True)
        max_out, _ = torch.max(cat, dim=1, keepdim=True)
        x = torch.cat([avg_out, max_out], dim=1) 
        x = self.conv(x) 
        return x

class ChannelAttentionafn(nn.Module):
    def __init__(self, channel, reduction=16):
        super(ChannelAttentionafn, self).__init__()
        self.avg_pool = nn.AdaptiveAvgPool2d(1)
        self.max_pool = nn.AdaptiveMaxPool2d(1)
        self.fc1 = nn.Conv2d(channel, channel // reduction, 1, bias=False)
        self.fc2 = nn.Conv2d(channel // reduction, channel, 1, bias=False)
        self.relu = nn.ReLU(inplace=True)
        self.activate = nn.Sigmoid()
    def forward(self, x):
        avg_out = self.relu(self.fc2(self.relu(self.fc1(self.avg_pool(x)))))
        max_out = self.relu(self.fc2(self.relu(self.fc1(self.max_pool(x)))))
        out = avg_out + max_out
        out = self.activate(out)
        return out

**Comment by ZZ-626** [comment]
Thank you very much
